internal/plugintest/working_dir_json_test.go

```diff
@@ -0,0 +1,69 @@
+package plugintest_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// TestJSONConfig verifies that TestStep.Config can contain JSON.
+// This test also proves that when changing the HCL and JSON formats back and
+// forth, the framework deletes the previous configuration file.
+func TestJSONConfig(t *testing.T) {
+	providerFactories := map[string]func() (*schema.Provider, error){
+		"tst": func() (*schema.Provider, error) { return tstProvider(), nil },
```

Comment:
Hopefully should appease the linter.

```suggestion
		"tst": func() (*schema.Provider, error) { return tstProvider(), nil }, //nolint:unparam // required signature
```